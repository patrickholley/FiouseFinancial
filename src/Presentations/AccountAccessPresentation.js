import React from 'react';
import { Animated, ScrollView, View } from 'react-native';
import styled from 'styled-components';
import { Actions } from 'react-native-router-flux';
import FButton from '../FiouseUI/FButton';
import FForm from '../FiouseUI/FForm';
import colors from '../constants/colors';

const AccountAccessView = styled.View`
  alignItems: center;
  display: flex;
  flex: 1;
  textAlign: center;
`;

const CopyrightText = styled.Text`
  margin: 10px;
`;

const FooterView = styled.View`
  flex: 1;
`;

const HeaderText = styled.Text`
  color: ${colors[0]};
  fontSize: 24;
  fontWeight: bold;
`;

const LargeLogoImage = styled.Image`
  borderRadius: 150;
  height: 150;
  margin: 10px;
`;

const SmallLogoImage = styled.Image`
  borderRadius: 100;
  height: 100;
  margin: 10px;
`;

const SubheaderAnimatedText = Animated.createAnimatedComponent(styled.Text`
  color: ${props => props.error ? 'red' : colors[2]};
  fontSize: 16;
  fontStyle: italic;
  height: 40px;
  textAlign: center;
`);

export default class AccountAccessPresentation extends React.PureComponent {

  render() {
    const resetPasswordParams = {
      formType: 'resetPassword',
      hideNavBar: false,
      title: 'Reset Password',
    };
    const createAccountParams = {
      formType: 'createAccount',
      hideNavBar: false,
      title: 'Create New Account',
    };

    const { formValues, isLoginForm, onFieldChange, onFormSubmit } = this.props;

    return (
      <ScrollView contentContainerStyle={{flexGrow: 1}}>
        <AccountAccessView>
          {isLoginForm
            ? <LargeLogoImage
                source={require('../../assets/fiouse_logo.png')}
                resizeMode="center"
              />
            : <SmallLogoImage
                source={require('../../assets/fiouse_logo.png')}
                resizeMode="center"
              />}
          <HeaderText>
            {formValues.headerText}
          </HeaderText>
          <SubheaderAnimatedText
            error={formValues.error}
            style={{ opacity: this.props.fadeAnim }}
          >
            {formValues.subheaderText}
          </SubheaderAnimatedText>
          <FForm
            fields={formValues.fields}
            onFieldChange={onFieldChange}
            onFormSubmit={onFormSubmit}
            submitText={formValues.submitText}
            buttonStyles={{ width: '60%' }}
          />
          {isLoginForm ? [
            <FButton
              backgroundColor={colors[2]}
              textColor="white"
              text="Forgot your password?"
              buttonStyles={{
                marginBottom: 10,
                width: '60%',
              }}
              onPress={() => Actions.push('accountAccess', resetPasswordParams)}
            />,
            <FButton
              backgroundColor={colors[0]}
              textColor="white"
              text="Need an account?"
              buttonStyles={{
                width: '60%',
              }}
              onPress={() => Actions.push('accountAccess', createAccountParams)}
            />
          ] : <View />}
          <FooterView />
          <CopyrightText>
            Copyright {'\u00A9'} 2018 Fiouse
          </CopyrightText>
        </AccountAccessView>
      </ScrollView>
    );
  }
}
