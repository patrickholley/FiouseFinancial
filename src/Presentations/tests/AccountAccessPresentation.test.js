import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AccountAccessPresentation from '../AccountAccessPresentation';

configure({ adapter: new Adapter() });

const testFormValues = {
  fields: {
    email: { placeholder: 'Email', type: 'email' },
    password: { placeholder: 'Password', type: 'password' },
  },
  headerText: 'Financial Assistant',
  subheaderText: '',
  submitText: 'Login',
};

const dummyFn = () => {};

it('renders correctly', () => {
  const wrapper = shallow(
    <AccountAccessPresentation
      canSubmit
      fadeAnim={{}}
      formValues={testFormValues}
      isLoginForm
      onFieldChange={dummyFn}
      onFormSubmit={dummyFn}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
