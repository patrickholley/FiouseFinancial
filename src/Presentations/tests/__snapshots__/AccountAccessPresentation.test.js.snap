// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountAccessPresentation
    canSubmit={true}
    fadeAnim={Object {}}
    formValues={
      Object {
        "fields": Object {
          "email": Object {
            "placeholder": "Email",
            "type": "email",
          },
          "password": Object {
            "placeholder": "Password",
            "type": "password",
          },
        },
        "headerText": "Financial Assistant",
        "subheaderText": "",
        "submitText": "Login",
      }
    }
    isLoginForm={true}
    onFieldChange={[Function]}
    onFormSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Component)>
        <Styled(Image)
          resizeMode="center"
          source={
            Object {
              "process": [Function],
            }
          }
        />
        <Styled(Text)>
          Financial Assistant
        </Styled(Text)>
        <AnimatedComponent
          error={undefined}
          style={
            Object {
              "opacity": Object {},
            }
          }
        >
          
        </AnimatedComponent>
        <FForm
          canSubmit={true}
          fields={
            Object {
              "email": Object {
                "placeholder": "Email",
                "type": "email",
              },
              "password": Object {
                "placeholder": "Password",
                "type": "password",
              },
            }
          }
          onFieldChange={[Function]}
          onFormSubmit={[Function]}
          submitButtonStyles={
            Object {
              "width": "60%",
            }
          }
          submitText="Login"
          submitTextStyles={Object {}}
        />
        <FButton
          backgroundColor="#4E7082"
          buttonStyles={
            Object {
              "marginBottom": 10,
              "width": "60%",
            }
          }
          isDisabled={false}
          onPress={[Function]}
          text="Forgot your password?"
          textColor="white"
          textStyles={Object {}}
        />
        <FButton
          backgroundColor="#DCAF68"
          buttonStyles={
            Object {
              "width": "60%",
            }
          }
          isDisabled={false}
          onPress={[Function]}
          text="Need an account?"
          textColor="white"
          textStyles={Object {}}
        />
        <Styled(Component) />
        <Styled(Text)>
          Copyright© 2018 Fiouse
        </Styled(Text)>
      </Styled(Component)>,
      "contentContainerStyle": Object {
        "flexGrow": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Image)
            resizeMode="center"
            source={
              Object {
                "process": [Function],
              }
            }
          />,
          <Styled(Text)>
            Financial Assistant
          </Styled(Text)>,
          <AnimatedComponent
            error={undefined}
            style={
              Object {
                "opacity": Object {},
              }
            }
          >
            
          </AnimatedComponent>,
          <FForm
            canSubmit={true}
            fields={
              Object {
                "email": Object {
                  "placeholder": "Email",
                  "type": "email",
                },
                "password": Object {
                  "placeholder": "Password",
                  "type": "password",
                },
              }
            }
            onFieldChange={[Function]}
            onFormSubmit={[Function]}
            submitButtonStyles={
              Object {
                "width": "60%",
              }
            }
            submitText="Login"
            submitTextStyles={Object {}}
          />,
          Array [
            <FButton
              backgroundColor="#4E7082"
              buttonStyles={
                Object {
                  "marginBottom": 10,
                  "width": "60%",
                }
              }
              isDisabled={false}
              onPress={[Function]}
              text="Forgot your password?"
              textColor="white"
              textStyles={Object {}}
            />,
            <FButton
              backgroundColor="#DCAF68"
              buttonStyles={
                Object {
                  "width": "60%",
                }
              }
              isDisabled={false}
              onPress={[Function]}
              text="Need an account?"
              textColor="white"
              textStyles={Object {}}
            />,
          ],
          <Styled(Component) />,
          <Styled(Text)>
            Copyright© 2018 Fiouse
          </Styled(Text)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "center",
            "source": Object {
              "process": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Financial Assistant",
          },
          "ref": null,
          "rendered": "Financial Assistant",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "error": undefined,
            "style": Object {
              "opacity": Object {},
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "canSubmit": true,
            "fields": Object {
              "email": Object {
                "placeholder": "Email",
                "type": "email",
              },
              "password": Object {
                "placeholder": "Password",
                "type": "password",
              },
            },
            "onFieldChange": [Function],
            "onFormSubmit": [Function],
            "submitButtonStyles": Object {
              "width": "60%",
            },
            "submitText": "Login",
            "submitTextStyles": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "resetPassword",
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#4E7082",
            "buttonStyles": Object {
              "marginBottom": 10,
              "width": "60%",
            },
            "isDisabled": false,
            "onPress": [Function],
            "text": "Forgot your password?",
            "textColor": "white",
            "textStyles": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "createAccount",
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#DCAF68",
            "buttonStyles": Object {
              "width": "60%",
            },
            "isDisabled": false,
            "onPress": [Function],
            "text": "Need an account?",
            "textColor": "white",
            "textStyles": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Copyright© 2018 Fiouse",
          },
          "ref": null,
          "rendered": "Copyright© 2018 Fiouse",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Component)>
          <Styled(Image)
            resizeMode="center"
            source={
              Object {
                "process": [Function],
              }
            }
          />
          <Styled(Text)>
            Financial Assistant
          </Styled(Text)>
          <AnimatedComponent
            error={undefined}
            style={
              Object {
                "opacity": Object {},
              }
            }
          >
            
          </AnimatedComponent>
          <FForm
            canSubmit={true}
            fields={
              Object {
                "email": Object {
                  "placeholder": "Email",
                  "type": "email",
                },
                "password": Object {
                  "placeholder": "Password",
                  "type": "password",
                },
              }
            }
            onFieldChange={[Function]}
            onFormSubmit={[Function]}
            submitButtonStyles={
              Object {
                "width": "60%",
              }
            }
            submitText="Login"
            submitTextStyles={Object {}}
          />
          <FButton
            backgroundColor="#4E7082"
            buttonStyles={
              Object {
                "marginBottom": 10,
                "width": "60%",
              }
            }
            isDisabled={false}
            onPress={[Function]}
            text="Forgot your password?"
            textColor="white"
            textStyles={Object {}}
          />
          <FButton
            backgroundColor="#DCAF68"
            buttonStyles={
              Object {
                "width": "60%",
              }
            }
            isDisabled={false}
            onPress={[Function]}
            text="Need an account?"
            textColor="white"
            textStyles={Object {}}
          />
          <Styled(Component) />
          <Styled(Text)>
            Copyright© 2018 Fiouse
          </Styled(Text)>
        </Styled(Component)>,
        "contentContainerStyle": Object {
          "flexGrow": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Image)
              resizeMode="center"
              source={
                Object {
                  "process": [Function],
                }
              }
            />,
            <Styled(Text)>
              Financial Assistant
            </Styled(Text)>,
            <AnimatedComponent
              error={undefined}
              style={
                Object {
                  "opacity": Object {},
                }
              }
            >
              
            </AnimatedComponent>,
            <FForm
              canSubmit={true}
              fields={
                Object {
                  "email": Object {
                    "placeholder": "Email",
                    "type": "email",
                  },
                  "password": Object {
                    "placeholder": "Password",
                    "type": "password",
                  },
                }
              }
              onFieldChange={[Function]}
              onFormSubmit={[Function]}
              submitButtonStyles={
                Object {
                  "width": "60%",
                }
              }
              submitText="Login"
              submitTextStyles={Object {}}
            />,
            Array [
              <FButton
                backgroundColor="#4E7082"
                buttonStyles={
                  Object {
                    "marginBottom": 10,
                    "width": "60%",
                  }
                }
                isDisabled={false}
                onPress={[Function]}
                text="Forgot your password?"
                textColor="white"
                textStyles={Object {}}
              />,
              <FButton
                backgroundColor="#DCAF68"
                buttonStyles={
                  Object {
                    "width": "60%",
                  }
                }
                isDisabled={false}
                onPress={[Function]}
                text="Need an account?"
                textColor="white"
                textStyles={Object {}}
              />,
            ],
            <Styled(Component) />,
            <Styled(Text)>
              Copyright© 2018 Fiouse
            </Styled(Text)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "center",
              "source": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Financial Assistant",
            },
            "ref": null,
            "rendered": "Financial Assistant",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "error": undefined,
              "style": Object {
                "opacity": Object {},
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "canSubmit": true,
              "fields": Object {
                "email": Object {
                  "placeholder": "Email",
                  "type": "email",
                },
                "password": Object {
                  "placeholder": "Password",
                  "type": "password",
                },
              },
              "onFieldChange": [Function],
              "onFormSubmit": [Function],
              "submitButtonStyles": Object {
                "width": "60%",
              },
              "submitText": "Login",
              "submitTextStyles": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "resetPassword",
            "nodeType": "class",
            "props": Object {
              "backgroundColor": "#4E7082",
              "buttonStyles": Object {
                "marginBottom": 10,
                "width": "60%",
              },
              "isDisabled": false,
              "onPress": [Function],
              "text": "Forgot your password?",
              "textColor": "white",
              "textStyles": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "createAccount",
            "nodeType": "class",
            "props": Object {
              "backgroundColor": "#DCAF68",
              "buttonStyles": Object {
                "width": "60%",
              },
              "isDisabled": false,
              "onPress": [Function],
              "text": "Need an account?",
              "textColor": "white",
              "textStyles": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Copyright© 2018 Fiouse",
            },
            "ref": null,
            "rendered": "Copyright© 2018 Fiouse",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
